### PR TITLE
util: Rework `get_host_addr`

### PR DESCRIPTION
**Pull request by xdustinface**
Start of a series of PR's working towards using

```
IPAddress = Union[IPv4Address, IPv6Address]
```

wherever we use string hosts except in some places where they are needed because it can be a domain also. One example would be in `PeerInfo` to use `address: IPAddress` instead of `host: str` which is a big example since it will require a bunch of other changes and will also be done in few steps but i think at the end if this is all done things will be much cleaner.

### CODE REVIEW

File: chia/util/network.py [hunk 1]
```diff
@@ -108,32 +109,25 @@ def class_for_type(type: NodeType) -> Any:
     raise ValueError("No class for type")
 
 
-def get_host_addr(host: Union[PeerInfo, str], prefer_ipv6: Optional[bool]) -> str:
-    # If there was no preference passed in (from config), set the system-wise
+def get_host_addr(host: str, prefer_ipv6: bool = False) -> IPAddress:
```
Comment (arvidn) [inline chia/util/network.py:112]: would it be really cumbersome to require this bool parameter to be named?

Comment (arvidn) [inline chia/util/network.py:112]:
```suggestion
def get_host_addr(host: str, *, prefer_ipv6: bool = False) -> IPAddress:
```

File: chia/util/network.py [hunk 2]
```diff
@@ -108,32 +109,25 @@ def class_for_type(type: NodeType) -> Any:
...
     for t in addrset:
         if prefer_ipv6 and t[0] == socket.AF_INET6:
-            return t[4][0]
+            return ip_address(t[4][0])
         if not prefer_ipv6 and t[0] == socket.AF_INET:
-            return t[4][0]
+            return ip_address(t[4][0])
     # If neither matched preference, just return the first available
-    return addrset[0][4][0]
+    return ip_address(addrset[0][4][0])
```
Comment (arvidn) [inline chia/util/network.py:130]: I realize you didn't really change this code, but it doesn't look correct. The list of IP addresses may contain multiple IPv4, multiple IPv6 and multiple of other address families. Just picking the first one here doesn't seem right. We should pick the first one that is either `AF_INET6` or `AF_INET`.

Maybe the loop should remember the first address of each type it finds, and then pick one of them that isn't `None` depending on the preference

File: chia/timelord/timelord_launcher.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def find_vdf_client() -> pathlib.Path:
     raise FileNotFoundError("can't find vdf_client binary")
 
 
-async def spawn_process(host: str, port: int, counter: int, lock: asyncio.Lock, prefer_ipv6: Optional[bool]):
+async def spawn_process(host: str, port: int, counter: int, lock: asyncio.Lock, prefer_ipv6: bool):
```
Comment (arvidn) [inline chia/timelord/timelord_launcher.py:42]:
```suggestion
async def spawn_process(host: str, port: int, counter: int, lock: asyncio.Lock, *, prefer_ipv6: bool):
```

Comment (arvidn) [inline chia/timelord/timelord_launcher.py:42]: would this break a lot of call sites?

**Review by arvidn — COMMENTED** [review]
looks generally good

File: chia/util/network.py (hunk 1 above)
Comment (xdustinface) [reply chia/util/network.py]: Funny that right now, all our call sites already refer to "prefer_ipv6" by either variable names or config fetches anyway but sure we can make it named, makes sense, see [d581b92](https://github.com/Chia-Network/chia-blockchain/pull/14068/commits/d581b9277fba2dbf7bed1ec56fb8d2e06024994d).

File: chia/timelord/timelord_launcher.py (hunk 3 above)
Comment (xdustinface) [reply chia/timelord/timelord_launcher.py]: Not more than i anyway already touched with this PR, so i also changed it in [d581b92](https://github.com/Chia-Network/chia-blockchain/pull/14068/commits/d581b9277fba2dbf7bed1ec56fb8d2e06024994d).

File: chia/util/network.py (hunk 2 above)
Comment (xdustinface) [reply chia/util/network.py]: Agree and yeah it makes sense to adjust this while reworking this method, see [bf72dbd](https://github.com/Chia-Network/chia-blockchain/pull/14068/commits/bf72dbd25a500cc7cbbeb5742b21328193c759af).

**Review by arvidn — APPROVED** [review]

**Comment by arvidn** [comment]
I think it might make sense to add the `cleanup` label to this PR. But if you don't, make sure the PR title makes sense in the changelog

**Pull request merged**